pkg/server/ca/manager/manager_test.go

```diff
@@ -587,12 +589,136 @@ func TestPruneBundle(t *testing.T) {
+		expectedCAJournals []*datastore.CAJournal
+		testJournals       []*testJournal
+	}{
+		{
+			name: "no journals with CAs expired before the threshold - no journals to be pruned",
+			testJournals: []*testJournal{
+				{
+					Journal: Journal{
+						config: jc,
+						entries: &journal.Entries{
+							X509CAs: []*journal.X509CAEntry{{NotAfter: now}, {NotAfter: tomorrow}},
+							JwtKeys: []*journal.JWTKeyEntry{{NotAfter: now}, {NotAfter: tomorrow}},
+						},
+					},
+				},
+				{
+					Journal: Journal{
+						config: jc,
+						entries: &journal.Entries{
+							X509CAs: []*journal.X509CAEntry{{NotAfter: now}, {NotAfter: tomorrow}},
+							JwtKeys: []*journal.JWTKeyEntry{{NotAfter: now}, {NotAfter: tomorrow}},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "some journals with CAs expired before the threshold, but not all - no journals to be pruned",
+			testJournals: []*testJournal{
+				{
+					Journal: Journal{
+						config: jc,
+						entries: &journal.Entries{
+							X509CAs: []*journal.X509CAEntry{{NotAfter: tomorrow}, {NotAfter: beforeThreshold}},
+							JwtKeys: []*journal.JWTKeyEntry{{NotAfter: beforeThreshold}, {NotAfter: tomorrow}},
+						},
+					},
+				},
+				{
+					Journal: Journal{
+						config: jc,
+						entries: &journal.Entries{
+							X509CAs: []*journal.X509CAEntry{{NotAfter: tomorrow}, {NotAfter: beforeThreshold}},
+							JwtKeys: []*journal.JWTKeyEntry{{NotAfter: beforeThreshold}, {NotAfter: tomorrow}},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "all CAs expired before the threshold in a journal - one journal to be pruned",
+			testJournals: []*testJournal{
+				{
+					shouldBePruned: true,
+					Journal: Journal{
+						config: jc,
+						entries: &journal.Entries{
+							X509CAs: []*journal.X509CAEntry{{NotAfter: beforeThreshold}, {NotAfter: beforeThreshold}},
+							JwtKeys: []*journal.JWTKeyEntry{{NotAfter: beforeThreshold}, {NotAfter: beforeThreshold}},
+						},
+					},
+				},
+				{
+					Journal: Journal{
+						config: jc,
+						entries: &journal.Entries{
+							X509CAs: []*journal.X509CAEntry{{NotAfter: tomorrow}, {NotAfter: beforeThreshold}},
+							JwtKeys: []*journal.JWTKeyEntry{{NotAfter: beforeThreshold}, {NotAfter: tomorrow}},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			// Have a fresh data store in each test case
+			test.ds = fakedatastore.New(t)
+			test.m.c.Catalog.(*fakeservercatalog.Catalog).SetDataStore(test.ds)
+
+			for _, j := range testCase.testJournals {
+				entriesBytes, err := proto.Marshal(j.entries)
+				require.NoError(t, err)
+				caJournal, err := test.ds.SetCAJournal(ctx, &datastore.CAJournal{
+					ActiveX509AuthorityID: "",
+					Data:                  entriesBytes,
+				})
+				require.NoError(t, err)
+
+				if !j.shouldBePruned {
+					testCase.expectedCAJournals = append(testCase.expectedCAJournals, caJournal)
```

Comment:
Since we need to populate the datastore with journal data, that's saved marshalled and each record is assigned an ID, I found easier to just build the expectedCAJournals list with the items returned from the SetCAJournal operation that we know that should not be pruned.

Reply:
Moved expectedCAJournals to a var.